### PR TITLE
[CO-380] Removing Migration.hs Migration/Types.hs and Migration/Monad…

### PR DESCRIPTION
**Pull request by paweljakubas**
## Description

Remove Migration.hs, Migration/Types.hs and Migration/Monad.hs from wallet-new.
See ticket below for more preliminary information.

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CO-380

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [x] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by paweljakubas** [comment]
I look at logs of walletIntegration and see https://hydra.iohk.io/build/265755/log:

**Comment by KtorZ** [comment]
@paweljakubas Indeed, I had a similar issue yesterday with the CI being finished but not correctly reporting it. 
ops are on it as far as I know :+1:

File: wallet-new/src/Cardano/Wallet/API/WIP/LegacyHandlers.hs [hunk 1]
```diff
@@ -20,51 +27,140 @@ import qualified Pos.Wallet.Web.State as V0 (WalletSnapshot, askWalletSnapshot)
...
+
+fromAssuranceLevelEither :: V1.AssuranceLevel -> Either MigrationError V0.CWalletAssurance
+fromAssuranceLevelEither V1.StrictAssurance = pure V0.CWAStrict
+fromAssuranceLevelEither V1.NormalAssurance = pure V0.CWANormal
+
+fromAssuranceLevel :: Catch.MonadThrow m  => V1.AssuranceLevel -> m V0.CWalletAssurance
+fromAssuranceLevel from = case fromAssuranceLevelEither from of
+    Left e   -> Catch.throwM e
+    Right to -> pure to
+
```
Comment (KtorZ) [inline wallet-new/src/Cardano/Wallet/API/WIP/LegacyHandlers.hs]: Minor, we can change on the story-branch directly when merging but `MonadThrow` is part of `Universum` so there's no need for the import above, and even less need for a qualified one ;) ... 
As a rule of thumb, prefer explicit imports over qualified ones. Used qualified imports when functions are ambiguous / context-related. For instance: `fromList`, `encode`, `toString` ... In such case, we'd prefer:

`Map.fromList`, `Yaml.encode` and `Scientific.toString` for example.

Comment (paweljakubas) [reply wallet-new/src/Cardano/Wallet/API/WIP/LegacyHandlers.hs]: Right!!! I can make this change in PR concerning CO-383 :-)

**Comment by KtorZ** [comment]
Merged manually in the base branch with the above change :+1:

**Pull request closed without merging**